modules/diskq/qdisk.c

```diff
@@ -777,6 +777,14 @@ _update_header_with_default_values(QDisk *self)
   self->hdr->backlog_len = 0;
 }
 
+static gboolean
+_create_path(const gchar *filename)
+{
+  FilePermOptions fpermoptions;
+  file_perm_options_defaults(&fpermoptions);
+  return file_perm_options_create_containing_directory(&fpermoptions, filename);
```

Comment:
We have global `dir-owner()`, `dir-group()`, and `dir-perm()` options.

Shouldn't we inherit them (in case they are set) with
`file_perm_options_inherit_from(&fpermoptions, &cfg->file_perm_options)`?

Reply:
`create-dirs()` is not taken into account either, If I understand it correctly, we just want disk-buffer directory creation to work without complicating things with these options.

Reply:
> `create-dirs`

Creating the path if not exists is something that make sense almost always (maybe there are some strange use-cases where it is not), and as such doing it unconditionally is fine.

> We have global dir-owner(), dir-group(), and dir-perm() options.

I was considering adding support for owner/group/perm. Just inheriting those globals is no good, in that case we should provide a way to set it locally. Also the disk queue file itself should inherit it as well (as it does not currently).
At this point I was thinking that addition is nice, but should be a new patchset and not in strict scope. That can be done always later.